go/engine/favorite_test.go

```diff
@@ -14,24 +16,25 @@ import (
 func TestFavoriteAdd(t *testing.T) {
 	tc := SetupEngineTest(t, "template")
 	defer tc.Cleanup()
-	CreateAndSignupFakeUser(tc, "fav")
+	u := CreateAndSignupFakeUser(tc, "fav")
+	expectedFaves := newFavorites(u.Username)
 
 	idUI := &FakeIdentifyUI{}
-	addfav("t_alice,t_bob", true, true, idUI, tc)
-	if len(listfav(tc)) != 3 {
-		t.Errorf("favorites len: %d, expected 3", len(listfav(tc)))
+	addfav("t_alice,t_bob", true, true, idUI, tc, expectedFaves)
+	if !listfav(tc).Equal(*expectedFaves) {
+		t.Errorf("bad favorites")
 	}
 
 	// Add the same share again. The number shouldn't change.
-	addfav("t_alice,t_bob", true, true, idUI, tc)
-	if len(listfav(tc)) != 3 {
-		t.Errorf("favorites len: %d, expected 3", len(listfav(tc)))
+	addfav("t_alice,t_bob", true, true, idUI, tc, nil)
+	if !listfav(tc).Equal(*expectedFaves) {
```

Comment:
So this isn't quite the same test, as now the server could be returning t_alice,t_bob twice and listfav would still equal expectedFaves since Push sets a map.

Could Push fail if the key already exists?